src/main/kotlin/no/nav/familie/ef/mottak/service/SøknadTilGenereltFormatMapper.kt

```diff
@@ -0,0 +1,180 @@
+                .map { finnFelter(it) }
+                .flatten()
+        }
+        val parametere = konstruktørparametere(entitet)
+
+        val list =
+            parametere
+                .asSequence()
+                .map { finnSøknadsfelt(entitet, it) }
+                .filter { it.visibility == KVisibility.PUBLIC }
+                .mapNotNull { getFeltverdi(it, entitet) }
+                .map { finnFelter(it) } // Kall rekursivt videre
+                .flatten()
+                .toList()
+
+        if (entitet is Søknadsfelt<*>) {
+            if (entitet.verdi!! is Dokumentasjon) {
+                @Suppress("UNCHECKED_CAST")
+                return listOf(mapDokumentasjon(entitet as Søknadsfelt<Dokumentasjon>))
+            }
+            if (entitet.verdi!!::class in endNodes) {
+                return listOf(Feltformaterer.mapEndenodeTilUtskriftMap(entitet))
+            }
+            if (entitet.label == "Barna dine") {
+                return listOf(feltlisteMap(entitet.label, list, FeltType.TABLE_BARN))
+            }
+            if (entitet.label == "Vedlegg") {
+                return listOf(feltlisteMap(entitet.label, list, FeltType.VEDLEGG))
+            }
+            if (entitet.verdi is List<*>) {
+                val verdiliste = entitet.verdi as List<*>
+                if (verdiliste.isNotEmpty() && verdiliste.first() is String) {
+                    return listOf(Feltformaterer.mapEndenodeTilUtskriftMap(entitet))
+                }
+            }
+            return listOf(feltlisteMap(entitet.label, list))
+        }
+        return list
+    }
+
+    private fun mapDokumentasjon(entitet: Søknadsfelt<Dokumentasjon>): Map<String, *> = feltlisteMap(entitet.label, listOf(Feltformaterer.mapEndenodeTilUtskriftMap(entitet.verdi.harSendtInnTidligere)))
+
+    private fun feltlisteMap(
+        label: String,
+        verdi: List<*>,
+        type: FeltType? = null,
+    ): Map<String, Any> =
+        if (type == null) {
+            mapOf("label" to label, "verdiliste" to verdi)
+        } else {
+            mapOf("label" to label, "type" to type.typeName, "verdiliste" to verdi)
+        }
+
+    /**
+     * Henter ut verdien for felt på entitet.
+     */
+    private fun getFeltverdi(
+        felt: KProperty1<out Any, Any?>,
+        entitet: Any,
+    ) = felt.getter.call(entitet)
+
+    /**
+     * Finn første (og eneste) felt på entiteten som har samme navn som konstruktørparameter.
+     */
+    private fun finnSøknadsfelt(
+        entity: Any,
+        konstruktørparameter: KParameter,
+    ) = entity::class.declaredMemberProperties.first { it.name == konstruktørparameter.name }
+
+    /**
+     * Konstruktørparametere er det eneste som gir oss en garantert rekkefølge for feltene, så vi henter disse først.
+     */
+    private fun konstruktørparametere(entity: Any) = entity::class.primaryConstructor?.parameters ?: emptyList()
+}
+
+enum class FeltType(
+    val typeName: String,
```

Comment:
I kontekst av `VisningsVariant` er `name` bra nok for meg
```suggestion
    val name: String,
```

Reply:
fikk en conflikt men en annen type